lib/kitchen/transport/base.rb

```diff
@@ -139,10 +139,15 @@ def execute_with_retry(command, retryable_exit_codes = [], max_retries = 1, wait
           end
         end
 
-        def retry?(current_try, max_retries, retryable_exit_codes, exit_code)
-          current_try <= max_retries &&
-            !retryable_exit_codes.nil? &&
-            retryable_exit_codes.flatten.include?(exit_code)
+        def retry?(current_try, max_retries, retryable_exit_codes, exception)
+          case exception
+          when Kitchen::Transport::TransportFailed
+            current_try <= max_retries &&
+              !retryable_exit_codes.nil? &&
+              retryable_exit_codes.flatten.include?(exception.exit_code)
+          else
+            false
+          end
```

Comment:
this might read slightly better as

```
   if exception.is_a?(Kitchen::Transport::TransportFailed)
     return [.... multiline conditional... ]
  end
  return false
```